Mathlib/Analysis/Normed/Affine/AsymptoticCone.lean

```diff
@@ -0,0 +1,428 @@
+/-
+Copyright (c) 2025 Attila Gáspár. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Attila Gáspár
+-/
+
+import Mathlib.Analysis.Convex.Between
+import Mathlib.Analysis.Normed.Module.FiniteDimension
+
+/-!
+# Asymptotic cone of a set
+
+This file defines the asymptotic cone of a set in a normed affine space. The asymptotic cone of a
+set $A$ is usually defined as the set of points $v$ for which there exist sequences $t_n > 0$ and
+$x_n \in A$ such that $t_n \to 0$ and $t_n x_n \to v$. We take a different approach here using
+filters: we define the asymptotic cone of `s` as the set of vectors `v` such that
+`∃ᶠ p in Filter.atTop • 𝓝 v, p ∈ s` holds.
```

Comment:
The second sentence onwards belongs in its own `## Implementation notes` section.